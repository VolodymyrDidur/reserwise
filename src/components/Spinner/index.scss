.spinner {
  $size: 50px;

  $tik-top: 2px;
  $tik-width: 4px;
  $tik-height: 12px;

  display: inline-block;
  @include set-size($size,$size);

  div {
    transform-origin: $size*0.5 $size*0.5;
    animation: spinner 1.2s linear infinite;

    &:after {
      content: ' ';
      display: block;
      position: absolute;
      top: $tik-top;
      left: $size*0.5 - $tik-width*0.5;
      width: $tik-width;
      height: $tik-height;
      border-radius: 20%;
      background: $color-purple;
    }

    @for $i from 0 through 11 {
      &:nth-child(#{$i + 1}) {
        transform: rotate($i * 30deg);
        animation-delay: #{-1.1 + 0.1 * $i}s;
      }
    }
  }
}

@keyframes spinner {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
