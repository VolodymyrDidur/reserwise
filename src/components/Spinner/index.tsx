import React from 'react';

import './index.scss';

const Spinner = () => {
  return (
    <div className="spinner">
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  );
};

export default Spinner;
