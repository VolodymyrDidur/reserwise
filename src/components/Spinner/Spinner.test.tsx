import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from './index';

const wrapper = shallow(<Spinner />);

describe('Spinner UI component', () => {
  it('Component match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('main component class exists', () => {
    expect(wrapper.find('.spinner').exists()).toBeTruthy();
  });
});
