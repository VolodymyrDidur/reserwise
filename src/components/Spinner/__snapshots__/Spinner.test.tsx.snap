// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner UI component Component match snapshot 1`] = `
<div
  className="spinner"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
